### PR TITLE
ECS fields that are arrays should use C# arrays

### PR DESCRIPTION
**Pull request by codebrain**
Relates to: https://github.com/elastic/ecs/pull/727

### CODE REVIEW

File: src/Elastic.CommonSchema/Serialization/BaseJsonFormatter.Generated.cs [hunk 1]
```diff
@@ -19,9 +19,9 @@ internal partial class BaseJsonConverter<TBase> : EcsJsonConverterBase<TBase>
 		where TBase : Base, new()
 	{
 		private static bool ReadProperties(
-			ref Utf8JsonReader reader,
-			TBase ecsEvent,
-			ref DateTimeOffset? timestamp,
+			ref Utf8JsonReader reader,
```
Comment (Mpdreamz) [inline src/Elastic.CommonSchema/Serialization/BaseJsonFormatter.Generated.cs]: Extra space, code cleanup should catch this but wonder how these were introduced (not important right now)

File: src/Generator/Schema/Field.cs [hunk 2]
```diff
@@ -31,19 +31,38 @@ public string ClrType
...
-				if (Name.Contains("supported_ciphers")) return "string[]";
-				if (Schema.Name == "vulnerability" && Name == "category") return "string[]";
-				if (Schema.Name == "file" && Name == "attributes") return "string[]";
-				if (Schema.Name == "dns" && Name == "header_flags") return "string[]";
-				if (Schema.Name == "dns" && Name == "resolved_ip") return "string[]";
-				if (Schema.Name == "user" && FlatName == "user.id") return "string[]";
-				if (Schema.Name == "base" && Name == "tags") return "string[]";
-				if (Schema.Name == "base" && Name == "labels") return "IDictionary<string, object>";
-				if (Schema.Name == "base" && Name == "_metadata") return "IDictionary<string, object>";
+				if (FlatName == "container.image.tag") return "string[]";
```
Comment (Mpdreamz) [inline src/Generator/Schema/Field.cs]: switch expression :)

File: src/Generator/Views/BaseJsonFormatter.Generated.cshtml [hunk 3]
```diff
@@ -67,7 +67,7 @@ namespace Elastic.CommonSchema.Serialization
 			};
 		}
 
-		public override void Write(Utf8JsonWriter writer, Base value, JsonSerializerOptions options)
+		public override void Write(Utf8JsonWriter writer, TBase value, JsonSerializerOptions options)
```
Comment (Mpdreamz) [inline src/Generator/Views/BaseJsonFormatter.Generated.cshtml]: ++

File: src/Generator/Schema/Field.cs (hunk 2 above)
Comment (codebrain) [reply src/Generator/Schema/Field.cs]: No doubt, it is ugly, I agree - when https://github.com/elastic/ecs/pull/727 lands this whole section will be deleted...

**Pull request merged**